File: 251003/Lysiakoff/rvlab5/rvlab4_publisher/src/main/java/com/publisher/rvlab4_publisher/mapper/MarkMapper.java
package com.publisher.rvlab4_publisher.mapper;

import com.publisher.rvlab4_publisher.dto.MarkRequestTo;
import com.publisher.rvlab4_publisher.dto.MarkResponseTo;
import com.publisher.rvlab4_publisher.entity.Mark;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public interface MarkMapper {

    MarkMapper INSTANCE = Mappers.getMapper(MarkMapper.class);

    Mark toEntity(MarkRequestTo markRequestTo);

    MarkResponseTo toResponseDto(Mark mark);

    Mark toEntity(MarkResponseTo markResponseTo);

    MarkRequestTo toRequestDto(Mark mark);
}

File: 251003/Kukhatskavolets/microblog/publisher/src/main/kotlin/by/kukhatskavolets/publisher/mappers/MarkMapper.kt
package by.kukhatskavolets.publisher.mappers

import by.kukhatskavolets.publisher.dto.requests.MarkRequestTo
import by.kukhatskavolets.publisher.dto.responses.MarkResponseTo
import by.kukhatskavolets.publisher.entities.Mark

fun MarkRequestTo.toEntity() = Mark(
    id = this.id,
    name = this.name
)

fun Mark.toResponse() = MarkResponseTo(
    id = this.id,
    name = this.name
)